### PR TITLE
Better Handle JSON Configuration Exceptions

### PR DESCRIPTION
**Pull request by mjenglish**
Previously, when running the program with an invalid JSON configuration file, the loading of the JSON failed silently and the user was left clueless.

In this pull request, I suggest catching specific exception types and handling them with the appropriate logic. 

Now, given an invalid JSON input, the logger displays clear error messages to the user of what went wrong.

### CODE REVIEW

**Comment by jskazinski** [comment]
One of the ramifications of this suggested change is that the RedfishConfig() no longer catches any exceptions thrown, so all callers of RedfishConfig() will now have to catch any exceptions and this would mean changes to 5 other callers of RedfishConfig(). So in this case, handling the exception in a single place is much cleaner that requiring all callers to handle exceptions.
Another change is the inclusion of a prompt to read input from stdin using the input() function which results in no longer being able to run automatic test cases which would pause forever or until timed out, diminishing the value of this tool for executing Redfish commands.
Having a better error message, and configuration file validation, would be an improvement.

**Comment by jskazinski** [comment]
As a practice with each code change, this project is following semantic versioning, so the version.py and CHANEGLOG.md should be updated with every pull request.

**Comment by mjenglish** [comment]
@jskazinski Thanks for the feedback.

I forgot about the `CHANGELOG.md`. I will add that.

Regarding your other comments, we will **not** merge this code in its current state, but I am going to keep this PR open and try to incorporate your feedback when I have time.

**Comment by mjenglish** [comment]
> Having a better error message, and configuration file validation, would be an improvement.

@jskazinski I updated this pull request. Could you please take another look?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Change Log
 
+## [2.4.2] - 2022-10-06
```
Comment (jskazinski) [inline CHANGELOG.md:3]: Please change version.py to 2.4.2?

**Review by jskazinski — APPROVED** [review]
Changes look good to me. Thanks for the improvement. Please update the version.py file also and feel free to merge.

File: CHANGELOG.md (hunk 1 above)
Comment (mjenglish) [reply CHANGELOG.md:3]: Sorry. It's done now.

**Pull request merged**